code/modules/food_and_drinks/food/egg.dm

```diff
@@ -13,13 +13,40 @@
 	name = "egg"
 	desc = "An egg!"
 	icon_state = "egg"
+	icon = 'code/modules/ranching/icons/eggs.dmi'
 	food_reagents = list(/datum/reagent/consumable/eggyolk = 5)
 	microwaved_type = /obj/item/food/boiledegg
 	foodtypes = MEAT | RAW
 	var/static/chick_count = 0 //I copied this from the chicken_count (note the "en" in there) variable from chicken code.
 	ant_attracting = FALSE
 	decomp_type = /obj/item/food/egg/rotten
 	decomp_req_handle = TRUE //so laid eggs can actually become chickens
+	bite_consumption = 100 // whole ass egg all at once
+
+	///the amount the chicken is grown
+	var/amount_grown = 0
+	///the type of chicken that layed this egg
+	var/mob/living/simple_animal/chicken/layer_hen_type
+	///happiness of the chicken
+	var/happiness = 0
+	///list of consumed food
+	var/list/consumed_food
+	///list of consumed reagents
+	var/list/consumed_reagents
+	///list of all possible mutations
+	var/list/mutations = list()
+	///eggs ore type
+	var/obj/item/stack/ore/production_type = null
+	///list of picked mutations should only ever be one
+	var/list/possible_mutations = list()
```

Comment:
because they need to be passed into the egg for hatch timers as i can't check parent because i want it to be a snapshot of when the egg was layed its kinda jank but i can't think of a better way